ui/src/app/components/resources.tsx

```diff
@@ -24,32 +25,48 @@ export const ResourceCardMenu: React.FunctionComponent<{
       switchCase(resourceType,
```

Comment:
nit: There's a bit of repetition here. It would be cleaner I think to externalize these fragments so the `switchCase` is more readable. Like I can read code with a bit of UI in it, or UI with a bit of code in it, but code containing UI that contains code seems error prone to me. Maybe a little function that converts icon, click handler, disabled, and text into a MenuItem, and another function that takes a list of MenuItems to make a React.Fragment. 

Or even have a little component just for this. There are opportunities for individual menu modes or menu items to be . inconsistent during maintenance with the current approach.

Reply:
You're right. If this is <30m of effort in my update-all-the-tests PR, I'll break it out.

Reply:
This is going to be nontrivial - the best way to do it would be along the lines of #2327. @ericsong is making a backlog tech debt ticket to convert non-notebook resource card types to this paradigm.

Reply:
Awesome, thanks for the update.